prisma/schema.prisma

```diff
@@ -121,9 +121,11 @@ model User {
   favorites          User[]         @relation("Favorites")
   sentRequests       Request[]      @relation("sentRequests")
   receivedRequests   Request[]      @relation("receivedRequests")
-  carpools           CarpoolGroup[] @relation("userCarpools")
+  carpool            CarpoolGroup?   @relation(fields: [carpoolId], references: [id])
+  carpoolId          String?
```

Comment:
Can you explain why this change was made? am i correct in saying that carpool will hold the IDs of all the CarpoolGroups that the User is part of, then carpoolD contains the id of the CarpoolGroup that the user actually joins?

Reply:
A User can only be part of a single CarpoolGroup. When a User is already part of a CarpoolGroup, the fields carpool and carpoolId will exist on this person. When they are not yet part of a CarpoolGroup, these properties will not exist. Earlier, for some reason that Owen and I are not really sure, we were in fact saving an array of carpools. That was really not what we wanted to do and I think it just slipped through without any of us noticing/questioning the change.

Reply:
So is carppolId a derived attribute in that it is simply the id of the carpool that the user maybe has? If so, wouldn't it make sense to just remove the carpoolId attribute and in our code we can access it by just user.carpool.id?

Reply:
The carpoolId field is required in the User modal to create a relation between the User and Carpool Models. We can't really do away with it.